kubepy/appliers.py

```diff
@@ -174,8 +177,26 @@ def raise_if_failed(self):
                 self._raise_for_failed_condition(condition)
         failures = self.status.get('failed', 0)
         if self.max_retries and failures > self.max_retries:
+            failed_pod = self.get_job_pod()
+            if failed_pod:
+                stdout, stderr = self.get_failed_pod_logs(failed_pod)
+            else:
+                stdout, stderr = '', ''
+            for line in str(stdout).split('\\n'):
```

Comment:
It's a bit overcomplicated to work on raw `bytes` type. 
You can go with:

```
  for line in stdout.decode().splitlines():
```

I'd also drop prefixing every line with stdout. kubectl logs mixes stdout and stderr anyway ([28167](https://github.com/kubernetes/kubernetes/issues/28167)), so what you call `stderr` here is not caontainer's `stderr` but `kubectl logs`'s `stderr` :)
That would leave us with a clean:
```
  print(stdout.decode())
```